### PR TITLE
Remove git and address code review comments

### PR DESCRIPTION
**Pull request by anayden**
Sorry, this is a bit of a mess now: I addressed most of your comment from https://github.com/neuro-inc/mlops-demo-oss-dogs/pull/8 in this PR (I marked the addressed ones as `resolved` in the original PR).

Current version should work fine with pachyderm training and have simplified code and readme. Let me know what you think. I can also backport my fixes to the original PR if that's better - just let me know, it's not that difficult.

### CODE REVIEW

File: .neuro/live-s3.yml [hunk 1]
```diff
@@ -2,53 +2,6 @@ kind: live
...
-    context: $[[ flow.workspace ]]/
-    build_preset: cpu-medium
-  label_studio:
-    ref: image:/$[[ project.owner ]]/$[[ flow.flow_id ]]/label_studio:21.1.18
-    dockerfile: $[[ flow.workspace ]]/label_studio/label_studio.Dockerfile
-    context: $[[ flow.workspace ]]/label_studio
-    build_preset: cpu-medium
-
 jobs:
   prepare_remote_dataset:
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml:6]: Let's put this job def from here and pachy live to project.yaml as a job.

Same for deploy_inference_platform, locust, filebrowser, postgres.

File: .neuro/live-s3.yml [hunk 2]
```diff
@@ -75,22 +28,20 @@ jobs:
     pass_config: True
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml:28]: Let's put this job def config (except of bash lines) from here and pahy live to project.live as mixin and use here, adding the bash section only.

Same for label studio, jupyter.
WDYT?

File: config/pipeline.json [hunk 3]
```diff
@@ -4,25 +4,30 @@
...
   "transform": {
-    "image": "ghcr.io/neuro-inc/neuro-flow:21.11.2",
+    "image": "ghcr.io/neuro-inc/neuro-flow:22.4.2",
     "cmd": [
-      "bash",
-      "-c",
-      "cd /tmp; mkdir -p .neuro; echo $PROJECT_B64 | base64 -d > .neuro/project.yml; echo $BATCH_B64 | base64 -d > .neuro/training.yaml; echo $PARAMS_B64 | base64 -d > .neuro/params.yaml; neuro-flow --show-traceback bake training --meta-from-file .neuro/params.yaml --local-executor"
+      "/bin/bash"
+    ],
+    "stdin": [
```
Comment (YevheniiSemendiak) [inline config/pipeline.json:11]: awesome idea! 👍

**Review by YevheniiSemendiak — COMMENTED** [review]
Early bird comments. Still running, but wdyt about those ones?

File: AWS-S3.md [hunk 4]
```diff
@@ -95,7 +62,8 @@ neuro-flow run prepare_remote_dataset
   neuro-flow run train
   ```
 
-- You may follow the training process in the [MLFlow](https://www.mlflow.org/) server's Web UI using the provided link or open it again.
+- You may follow the training process in the [MLFlow](https://www.mlflow.org/) server's Web UI using the provided link
+  or open it again.
 
   ```shell
   neuro-flow run mlflow_server
```
Comment (YevheniiSemendiak) [inline AWS-S3.md:69]: maybe, let's put this line before training?

File: AWS-S3.md [hunk 5]
```diff
@@ -104,19 +72,24 @@ neuro-flow run prepare_remote_dataset
 - Pick a run ID value from MLFlow's Web UI and deploy the trained model as a REST API on the platform:
 
   ```shell
-  neuro-flow run deploy_inference_platform --param run_id XXXXXX  --param mlflow_storage $MLFLOW_STORAGE
+  neuro-flow run deploy_inference_platform --param run_id XXXXXX
```
Comment (YevheniiSemendiak) [inline AWS-S3.md:75]: we still require mlflow_storage parameter, let's either get rid of this param, or specify it in readme

File: .neuro/live-s3.yml [hunk 6]
```diff
@@ -298,7 +245,7 @@ jobs:
       PGDATA: /var/lib/postgresql/data/pgdata
 
   mlflow_server:
-    image: neuromation/mlflow:1.11.0
+    image: neuromation/mlflow:v1.25.1
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml]:
```suggestion
    image: ghcr.io/neuro-inc/mlflow:v1.25.2
```

File: .neuro/live-pachyderm.yml [hunk 7]
```diff
@@ -266,16 +218,20 @@ jobs:
       PGDATA: /var/lib/postgresql/data/pgdata
 
   mlflow_server:
-    image: neuromation/mlflow:1.11.0
+    image: neuromation/mlflow:v1.25.1
```
Comment (YevheniiSemendiak) [inline .neuro/live-pachyderm.yml]:
```suggestion
    image: ghcr.io/neuro-inc/mlflow:v1.25.2
```

File: .neuro/live-s3.yml [hunk 8]
```diff
@@ -169,8 +119,8 @@ jobs:
       neuro blob cp /label-studio/data/label_studio.sqlite3 blob:${BUCKET_NAME}
 
       # Validate dataset:
-      neuro blob ls -r blob:${BUCKET_NAME}
-      echo "Total images: $(neuro blob ls -r blob:${BUCKET_NAME} | grep -oP '\w*\.\w+$' | wc -l)" # grep files only
+      neuro blob ls -r blob:${BUCKET_NAME} | tee
+      echo "Total images: $(neuro blob ls -r blob:${BUCKET_NAME}/images/ | grep -oP '\w*\.\w+$' | wc -l)" # grep files only
 
   deploy_inference_platform:
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml:126]: pls add same default for `mlflow_storage` param as you did in pachy live:
`        default: 'storage:/${{ project.owner }}/${{ flow.project_id }}/mlruns'`

File: .neuro/live-s3.yml [hunk 9]
```diff
@@ -117,22 +68,21 @@ jobs:
...
       LABEL_STUDIO_LOCAL_FILES_SERVING_ENABLED: True
       LABEL_STUDIO_PROJECT: /label-studio-project
       LS_TOKEN: secret:ls-token
       LS_PORT: 443
     bash: |
-      git clone -b ${PROJECT_GIT_BRANCH} ${PROJECT_GIT_REPO} ${PROJECT}
-
       # Pull from S3
-      neuro blob cp blob:${BUCKET_NAME}/label_studio.sqlite3 "${PROJECT}/data/label_studio.sqlite3"
+      neuro blob cp blob:${BUCKET_NAME}/label_studio.sqlite3 "${PROJECT}/data/label_studio.sqlite3" || echo Label Studio data was not found
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml:82]:
```suggestion
      mkdir -p ${PROJECT}/data
      neuro blob cp blob:${BUCKET_NAME}/label_studio.sqlite3 "${PROJECT}/data/label_studio.sqlite3" || echo Label Studio data was not found
```
since now this directory does not exist

File: Pachyderm.md [hunk 10]
```diff
@@ -24,7 +24,8 @@ Build all images used in the flow:
 neuro-flow build ALL
 ```
 
-Create a secret with a private SSH key for accessing the GitHub repository (pull/push access should be allowed and the SSH key should not be protected with a passphrase):
+Create a secret with a private SSH key for accessing the GitHub repository (pull/push access should be allowed and the
```
Comment (YevheniiSemendiak) [inline Pachyderm.md]: Redundant.

File: Pachyderm.md [hunk 11]
```diff
@@ -33,7 +34,7 @@ neuro secret add gh-rsa @~/.ssh/id_rsa
 Create a service account for authenticating the training pipeline:
 
 ```shell
-neuro service-account create --name mlops-demo-oss-dogs
+neuro service-account create --name mlops_demo_oss_dogs
```
Comment (YevheniiSemendiak) [inline Pachyderm.md]: Sorry for my misslead in https://github.com/neuro-inc/mlops-demo-oss-dogs/pull/8#discussion_r845477127,
It [turned out](https://neuromation.slack.com/archives/CE9SZE5B3/p1649360440925959), the service account names with underscores are forbidden.
Let's revert this change.

File: Pachyderm.md [hunk 12]
```diff
@@ -52,34 +53,22 @@ Grant permissions for the new service account
 
 ```shell
 export USER=<YOUR_USERNAME>
-export PROJECT=mlops-demo-oss-dogs
+export PROJECT=mlops_demo_oss_dogs
 export PREFIX="/${USER}/${PROJECT}/"
 export ACCOUNT=${USER}/service-accounts/${PROJECT}
```
Comment (YevheniiSemendiak) [inline Pachyderm.md]:
```suggestion
export ACCOUNT=${USER}/service-accounts/mlops-demo-oss-dogs
```

File: Pachyderm.md [hunk 13]
```diff
@@ -52,34 +53,22 @@ Grant permissions for the new service account
...
 export ROLE=${USER}/projects/${PROJECT//-/_}
 
 neuro acl grant storage:${PREFIX} ${ROLE} write
 neuro acl grant job:/${ACCOUNT} ${ACCOUNT} manage
 neuro acl grant image:${PREFIX} ${ROLE} read
```
Comment (YevheniiSemendiak) [inline Pachyderm.md]: this line is redundant since the image is shared with the project role automatically by the Neuro-Flow

File: .neuro/live-s3.yml [hunk 14]
```diff
@@ -227,9 +177,8 @@ jobs:
       - ${{ params.mlflow_storage }}/0/${{ params.run_id }}/artifacts/model/data/model.h5:/storage/model.h5
     env:
       PYTHONPATH: /usr/project
-      PROJECT: /usr/project
       EXPOSE_SSH: "yes"
-      DATA_PATH: ${{env.PROJECT}}/data/Images
+      DATA_PATH: /usr/project/data/Images
```
Comment (YevheniiSemendiak) [inline .neuro/live-s3.yml:184]: In #Download data below here:
`neuro blob cp -rT blob:${BUCKET_NAME}/images ${DATA_PATH}`
if the bucket is emply, let's fail, since there is no sense to run jupyter

File: .neuro/live-pachyderm.yml [hunk 15]
```diff
@@ -285,49 +241,67 @@ jobs:
...
+      echo "Pipeline '${{ params.pachy_pipeline_name }}' created successfully."
+
+  train:
+    image: $[[ images.train.ref ]]
+    preset: cpu-medium
+    life_span: 1d
+    params:
+      mlflow_storage:
+        descr: Storage path, where MLFlow server stores model binaries
+        default: 'storage:/${{ project.owner }}/${{ flow.id }}/mlruns'
```
Comment (YevheniiSemendiak) [inline .neuro/live-pachyderm.yml]: `${{ flow.id }}` is deprecated, let's use `${{ flow.project_id }}`, instead https://neu-ro.gitbook.io/neuro-flow/reference/project-configuration-syntax#id or `${{ flow.flow_id }}`

File: .neuro/live-pachyderm.yml [hunk 16]
```diff
@@ -215,18 +167,18 @@ jobs:
...
+      DATA_PATH: /usr/project/data/Images
     params:
       run_id:
         descr: MLFlow run ID, which model should be deployed
       mlflow_storage:
         descr: Storage path, where MLFlow server stores model binaries
+        default: 'storage:/${{ project.owner }}/${{ flow.id }}/mlruns'
     bash: |
       # Init pachctl
       pachctl config update context default --pachd-address ${PACHY_URI}
```
Comment (YevheniiSemendiak) [inline .neuro/live-pachyderm.yml:184]: below, `pachctl get file ${PACHY_REPO}@master:/data/Images -r ${PROJECT}/data/Images | tee` -> `pachctl get file ${PACHY_REPO}@master:/data/Images -r -o ${PROJECT}/data/Images | tee`

**Review by YevheniiSemendiak — COMMENTED** [review]
Hey, I've finally finished the review, sorry for suuuuch a long time doing it.
In the majority, everything looks great, thank you for such cool improvements in this cumbersome project :)

the only issue I can see for now is the ambiguous usage of MLFlow, in some places we assume it is running under another's account, at some - under the project owner, but in general - all great.
Thank you <3

File: AWS-S3.md (hunk 4 above)
Comment (anayden) [reply AWS-S3.md:69]: If you open up MLFlow before running training, you won't see anything and I think this is misleading.

File: AWS-S3.md (hunk 5 above)
Comment (anayden) [reply AWS-S3.md:75]: Good catch. I had a default in the pachyderm yaml, but forgot to put it in the s3 yaml. Updated.

File: .neuro/live-s3.yml (hunk 14 above)
Comment (anayden) [reply .neuro/live-s3.yml:184]: Why not? One can run Jupyter for some other needs, such as running commands to populate data or just testing something

Comment (YevheniiSemendiak) [reply .neuro/live-s3.yml:184]: Yes, you are correct regarding running Jupyter.
Let's only change:
1. The command, which copies data from the bucket to `neuro blob cp -rT blob:${BUCKET_NAME}/images ${DATA_PATH}`, since currently the path to images in the notebook is broken.
Or we could change the notebook.

2. Add `pass_config` to the job description.

Sorry, I've mixed up two questions into one. My bad habit.

File: .neuro/live-s3.yml (hunk 9 above)
Comment (YevheniiSemendiak) [reply .neuro/live-s3.yml:82]: This one is also actual. Currently, when the LS starts, the directory `${PROJECT}/data` does not exist and this command fails even if `blob:${BUCKET_NAME}/label_studio.sqlite3` exists.
image

File: .neuro/live-s3.yml (hunk 14 above)
Comment (anayden) [reply .neuro/live-s3.yml:184]: Ok, done!

**Review by YevheniiSemendiak — APPROVED** [review]
LGTM! Thank you very much!
